mesonbuild/dependencies/dev.py

```diff
@@ -584,7 +584,7 @@ def __init__(self, environment: 'Environment', kwargs: JNISystemDependencyKW):
                 self.is_found = False
                 return
 
-        if 'version' in kwargs and not version_compare(self.version, kwargs['version']):
+        if 'version' in kwargs and not version_compare_many(self.version, kwargs['version']):
```

Comment:
version_compare_many returns a tuple with 3 elements. Therefore, it will always evaluate as True when converted to bool.